TESTS/mbed_drivers/timerevent/main.cpp

```diff
@@ -0,0 +1,352 @@
+    TEST_ASSERT_EQUAL(0, sem_slots);
+    tte.remove();
+
+    sem_slots = tte.sem_wait(TEST_DELAY_US * 2 / 1000 + 1);
+    TEST_ASSERT_EQUAL(0, sem_slots);
+}
+
+/** Test insert zero
+ *
+ * Given an instance of @a TimerEvent subclass initialized with
+ *     an empty @a ticker_queue structure
+ * When @a ticker_queue.present_time is greater than 0 us
+ *     and a tiestamp of 0 us is set with @a insert()
+ * Then object's internal event is scheduled with timestamp UINT32_MAX + 1
+ */
+void test_insert_zero(void) {
+    ticker_event_queue_t ticker_queue = { 0 };
+    ticker_interface_t ticker_interface = mock_ticker_interface;
+    ticker_interface.read = mock_ticker_read_const;
+    ticker_data_t ticker_data = {
+        .interface = &ticker_interface,
+        .queue = &ticker_queue
+    };
+    TestTimerEvent tte(&ticker_data);
+
+    tte.insert(0UL);
+    TEST_ASSERT_EQUAL_PTR(tte.get_event_addr(), ticker_queue.head);
+    TEST_ASSERT_EQUAL(1ULL << 32, ticker_queue.head->timestamp);
+    TEST_ASSERT_NULL(ticker_queue.head->next);
+
+    tte.remove();
+}
+
+/** Test insert_absolute zero
+ *
+ * Given an instance of @a TimerEvent subclass
+ * When a timestamp of 0 us is set with @a insert_absolute()
+ * Then an event handler is called instantly
+ */
+void test_insert_absolute_zero(void) {
+    TestTimerEvent tte;
+    tte.sem_wait(0);
+
+    tte.insert_absolute(0ULL);
+    int32_t sem_slots = tte.sem_wait(0);
+    TEST_ASSERT_EQUAL(1, sem_slots);
+
+    tte.remove();
+}
+
+/** Test insert timestamp from the past
+ *
+ * Given an instance of @a TimerEvent subclass initialized with
+ *     an empty @a ticker_queue structure
+ * When a timestamp of X us is set with @a insert()
+ *     and X is less than ticker_queue.present_time
+ * Then object's internal event is scheduled with timestamp X + UINT32_MAX + 1
+ */
+void test_insert_past(void) {
+    ticker_event_queue_t ticker_queue = { 0 };
+    ticker_interface_t ticker_interface = mock_ticker_interface;
+    ticker_interface.read = mock_ticker_read_const;
+    ticker_data_t ticker_data = {
+        .interface = &ticker_interface,
+        .queue = &ticker_queue
+    };
+    TestTimerEvent tte(&ticker_data);
+    const timestamp_t ts = MOCK_TICKER_CNT - 1;
+
+    tte.insert(ts);
+    TEST_ASSERT_EQUAL_PTR(tte.get_event_addr(), ticker_queue.head);
+    TEST_ASSERT_EQUAL_UINT64(ts + (1ULL << 32), ticker_queue.head->timestamp);
+    TEST_ASSERT_NULL(ticker_queue.head->next);
+
+    tte.remove();
+}
+
+/** Test insert_absolute timestamp from the past
+ *
+ * Given an instance of @a TimerEvent subclass
+ * When a timestamp of X us is set with @a insert_absolute()
+ *     and X is less than present_time
+ * Then an event handler is called instantly
+ */
+void test_insert_absolute_past(void) {
+    TestTimerEvent tte;
+    tte.sem_wait(0);
+
+    tte.insert_absolute(tte.ticker_read_us() - 1ULL);
+    int32_t sem_slots = tte.sem_wait(0);
+    TEST_ASSERT_EQUAL(1, sem_slots);
```

Comment:
Until #5051 is solved this assertion will fail on _NUCLEO_F070RB_ & _EFM32GG_STK3700_.